src/sinks/clickhouse/arrow_schema.rs

```diff
@@ -0,0 +1,767 @@
+        return Err(format!(
+            "Expected parentheses around arguments in '{}'",
+            input
+        ));
+    }
+
+    let inner = trimmed[1..trimmed.len() - 1].trim();
+    if inner.is_empty() {
+        return Ok(vec![]);
+    }
+
+    // Split by comma, handling nested parentheses and quotes
+    let mut args = Vec::new();
+    let mut current_arg = String::new();
+    let mut depth = 0;
+    let mut in_quotes = false;
+
+    for c in inner.chars() {
+        match c {
+            '\'' if !in_quotes => in_quotes = true,
+            '\'' if in_quotes => in_quotes = false,
+            '(' if !in_quotes => depth += 1,
+            ')' if !in_quotes => depth -= 1,
+            ',' if depth == 0 && !in_quotes => {
+                args.push(current_arg.trim().to_string());
+                current_arg = String::new();
+                continue;
+            }
+            _ => {}
+        }
+        current_arg.push(c);
+    }
+
+    if !current_arg.trim().is_empty() {
+        args.push(current_arg.trim().to_string());
+    }
+
+    Ok(args)
+}
+
+/// Parses ClickHouse Decimal types and returns the appropriate Arrow decimal type.
+/// ClickHouse formats:
+/// - Decimal(P, S) -> generic decimal with precision P and scale S
+/// - Decimal32(S) -> precision up to 9, scale S
+/// - Decimal64(S) -> precision up to 18, scale S
+/// - Decimal128(S) -> precision up to 38, scale S
+/// - Decimal256(S) -> precision up to 76, scale S
+///
+/// Uses metadata from ClickHouse's system.columns when available, otherwise falls back to parsing the type string.
+fn parse_decimal_type(ch_type: &str) -> Result<DataType, String> {
+    // Parse from type string
+    let (type_name, args_str) = extract_identifier(ch_type);
+
+    let result = parse_args(args_str).ok().and_then(|args| match type_name {
+        "Decimal" if args.len() == 2 => args[0].parse::<u8>().ok().zip(args[1].parse::<i8>().ok()),
+        "Decimal32" | "Decimal64" | "Decimal128" | "Decimal256" if args.len() == 1 => {
+            args[0].parse::<i8>().ok().map(|scale| {
+                let precision = match type_name {
+                    "Decimal32" => DECIMAL32_PRECISION,
+                    "Decimal64" => DECIMAL64_PRECISION,
+                    "Decimal128" => DECIMAL128_PRECISION,
+                    "Decimal256" => DECIMAL256_PRECISION,
+                    _ => unreachable!(),
+                };
+                (precision, scale)
+            })
+        }
+        _ => None,
+    });
+
+    result
+        .map(|(precision, scale)| {
+            if precision <= DECIMAL128_PRECISION {
+                DataType::Decimal128(precision, scale)
+            } else {
+                DataType::Decimal256(precision, scale)
+            }
+        })
+        .ok_or_else(|| {
+            format!(
+                "Could not parse Decimal type '{}'. Valid formats: Decimal(P,S), Decimal32(S), Decimal64(S), Decimal128(S), Decimal256(S)",
```

Comment:
```suggestion
                "Could not parse Decimal type '{}'. Valid formats: Decimal(P, S), Decimal32(S), Decimal64(S), Decimal128(S), Decimal256(S)",
```

Maintain consistency with docstring